sdk/python/kfp/dsl/for_loop.py

```diff
@@ -301,21 +377,30 @@ def __init__(
         output: pipeline_channel.PipelineChannel,
     ) -> None:
         self.output = output
+        # we know all dsl.Collected instances are lists, so set `is_artifact_list``
+        # for type checking, which occurs before dsl.Collected is updated to
+        # it's "correct" channel during compilation
         if isinstance(output, pipeline_channel.PipelineArtifactChannel):
             channel_type = output.channel_type
             self.is_artifact_channel = True
-            # we know all dsl.Collected instances are lists, so set to true
-            # for type checking, which occurs before dsl.Collected is updated to
-            # it's "correct" channel during compilation
-            self.is_artifact_list = True
+            is_artifact_list = True
+        elif isinstance(output, pipeline_channel.PipelineChannel
```

Comment:
Can we simplify by checking if `isinstance(output, pipeline_channel.PipelineParameterChannel)`?

Reply:
Done.